### ISSUE TITLE
Excluded RuboCop rules still show up in VSCode?

### ISSUE THREAD

**Issue by thomthom** [castwide/vscode-solargraph#57]
I see RuboCop warnings in the IDE for things that are disabled by the `.rubocop.yml` file. (In this case `.rubocop_todo.yml`:

```yml
# Offense count: 2
# Cop supports --auto-correct.
# Configuration parameters: ExactNameMatch, AllowPredicates, AllowDSLWriters, IgnoreClassMethods, Whitelist.
# Whitelist: to_ary, to_a, to_c, to_enum, to_h, to_hash, to_i, to_int, to_io, to_open, to_path, to_proc, to_r, to_regexp, to_str, to_s, to_sym
Style/TrivialAccessors:
  Exclude:
    - 'src/tt_truebend/bender.rb'
```

![image](https://user-images.githubusercontent.com/192418/40734659-58256a7c-6439-11e8-9dd6-f67699313b35.png)

But strangely enough, another setting do seem to work:

```yml
# Offense count: 8
# Configuration parameters: CountComments.
Metrics/MethodLength:
  Max: 99
```

Anyone run into this?

When I run rubocop from the console there are warnings.

(RuboCop 0.54, Solargraph 0.22, VSCode extension 0.17.4)

**Comment by castwide** [comment]
I just tried a couple tests. When I exclude a file from `Style/TrivialAccessors` in .rubocop.yml, it works as expected. However, when I use `inherit_from: .rubocop_todo.yml` in .rubocop.yml and put the exclude in .rubocop_todo.yml, it doesn't work. Is this the same behavior you're seeing?

From what I can tell, the problem is specific to rubocop commands that accept file content as input instead of reading from disk, which is necessary to diagnose the version of the file in the editor. I'm not sure how to fix it, but I'll keep looking into it.

**Comment by thomthom** [comment]
Yes, the exclude rule is coming from `inherit_from: .rubocop_todo.yml`. However, so is `Metrics/MethodLength:`. So it appear that `inherit_from: .rubocop_todo.yml`. is at least partially used.

So this might be an issue with RuboCop itself?

**Comment by thomthom** [comment]
Now I'm curious if other IDEs experience the same thing. I'll see if I can play around with RubyMine over the weekend - it should recently have added support for RuboCop for their linting.

And... the Ruby extension for VSCode... can compare that as well. Should be able to give us clues to whether this is a RuboCop issue or not.

**Comment by castwide** [comment]
I get the same behavior with solargraph disabled and rubocop enabled in vscode-ruby.

**Comment by thomthom** [comment]
Interesting.

I'm trying to compare against RubyMine, but right now I cannot get any RuboCop warning to appear there... I wanted to see if they had found a way.

**Comment by castwide** [comment]
I may have found the issue. It appears to be specific to Windows file paths. The command executed from Solargraph uses lower-case drive letters, e.g., `d:/path/to/file.rb`. When I force it to start with `D:`, it works.

**Comment by thomthom** [comment]
Really? How strange - didn't think Windows filesystem was case sensitive.

**Comment by castwide** [comment]
You're right, it's not. I suspect RuboCop's path comparisons are case-sensitive, and some method it uses to generate absolute paths uses capitalized drive letters.

As a workaround, I'll probably convert drive letters from the RuboCop reporter inside Solargraph. It's not perfect, but it seems like the easiest way to handle it without impacting other components.

**Comment by thomthom** [comment]
I just ran into another case where RuboCop handles file case sensitive, with it's `Exclude` config. I had typed all lower case. But some of the folders had capitalised naming. Probably would have taken me longer to work that out had it not been for this issue.

**Comment by castwide** [comment]
Yeah, I think we need to assume that path comparisons are always case-sensitive in RuboCop. The original issue here added another nuance to it because RuboCop assumes drive letters are upper-case, while VS Code assumes they're lower-case.

The drive letter fix is pushed to [castwide/solargraph](https://github.com/castwide/solargraph) master. It'll be included in the next gem release.

**Comment by thomthom** [comment]
Thanks for looking into that.

I searched the RuboCop issue tracker for case sensitivity issues. Though there wasn't much - and nothing on the topic of what we've been running into. I might start a thread on that. (Just need to get head above water.)

**Comment by castwide** [comment]
This issue should be resolved in gem 0.23. If not, please let me know.
